src/unit_tests/os_crypto/hmac/test_hmac_op.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stddef.h>
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include "../headers/shared.h"
+#include "../../os_crypto/hmac/hmac.h"
+
+/* setups/teardowns */
+static int setup_group(void **state) {
+    test_mode = 1;
+    return 0;
+}
+
+static int teardown_group(void **state) {
+    test_mode = 0;
+    return 0;
+}
+
+// Tests
+
+void test_hmac_string(void **state)
+{
+    const char *key = "test_key";
+    const char *string = "test string";
+    const char *string_hmac = "bcbf151b282a23e0849453f2b5732bfd8226e16a";
+
+    os_sha1 buffer;
+
+    OS_HMAC_SHA1_Str(key, string, buffer);
+
+    assert_string_equal(buffer, string_hmac);
+}
+
+void test_hmac_file(void **state)
+{
+    const char *key = "test_key";
+    const char *string = "test string";
+    const char *string_hmac = "bcbf151b282a23e0849453f2b5732bfd8226e16a";
+
+    os_sha1 buffer;
+
+    /* create tmp file */
+    char file_name[256];
+    strncpy(file_name, "/tmp/tmp_file-XXXXXX", 256);
+    int fd = mkstemp(file_name);
+
+    write(fd, string, strlen(string));
+    close(fd);
+
+    OS_HMAC_SHA1_File(key, file_name, buffer, OS_TEXT);
```

Comment:
The return value of the function is not asserted. It could be -1 in case that the file couldn't be opened.

Reply:
done